index.js

```diff
@@ -7,74 +7,109 @@
 
 'use strict';
 
+var define = require('define-property');
+
 /**
- * Copy properties from an object to another object,
- * where properties with function values will be
- * invoked in the context of the provider, and properties
- * with non-function values are just copied.
+ * Returns a function for copying properties from an object
+ * to another object.
  *
  * ```js
- * var receiver = {};
- * var provider = {
- *   name: 'provider',
- *   upper: function (str) {
- *     return '[' + this.name + '] ' + str.toUpperCase();
- *   }
- * };
- * var receiver = forward(receiver, provider);
- * receiver.name = 'receiver';
- * console.log(receiver.upper('foo'));
- * //=> [provider] FOO
+ * var forward = require('forward-object')({omit: ['foo']});
  * ```
  *
- * @param  {Object} `receiver` Object to receive properties.
- * @param  {Object} `provider` Object providing properties.
- * @param  {Array} `keys` Optional array of keys to foward.
- * @return {Object} Modified `receiver` object with properties from `provider`
+ * @param {Object} `options`
+ * @param {Boolean} `options.enumerable` Make copied properties enumerable. `true` is the default.
+ * @param {Array|String} `options.keys` Limit forwarding to the specified keys.
+ * @param {Array|String} `options.omit` Exclude the specified keys from being copied.
  * @api public
  */
 
-function forward(receiver, provider, keys) {
-  keys = keys || allKeys(provider);
-  keys = Array.isArray(keys) ? keys : [keys];
+module.exports = function(options) {
+  options = options || {};
+  var keys = options.keys;
+  var omit = options.omit || [];
+
+  /**
+   * Copy properties from the `provider` object to the `receiver`.
+   *
+   * Properties with function values will be invoked in the context
+   * of the provider, and properties with non-function values are
+   * just copied.
+   *
+   * @name forward
+   * @param {Object} `receiver`
+   * @param {Object} `provider`
+   * @return {Object} Receiver with forwarded properties from provider.
+   * @api public
+   */
 
-  keys.forEach(function (key) {
-    var val = provider[key];
+  return function forwardObject(receiver, provider) {
+    keys = keys ? arrayify(keys) : objectKeys(provider, omit);
```

Comment:
At first glance, it looks like if the `keys` aren't set originally and you call this function twice, then the second call will use the first call's `provider`'s keys.

Might just need to change to `var keys = options.keys ? arrayify(options.keys) : objectKeys(provider, omit);`